stubs/requests/requests/utils.pyi

```diff
@@ -22,6 +26,7 @@ def super_len(o): ...
 def get_netrc_auth(url: _Uri, raise_errors: bool = ...) -> tuple[str, str] | None: ...
 def guess_filename(obj): ...
 def extract_zipped_paths(path): ...
+def atomic_open(filename: PathLike[AnyStr]) -> AbstractContextManager[BufferedWriter]: ...
```

Comment:
- The `AnyStr` `TypeVar` is meaningless here, since it's only used once in the function signature.
- This function works with `str` or `bytes`, not just objects that have an `__fspath__` method
- You can be more specific in the return type -- you can use `contextlib._GeneratorContextManager` instead of `AbstractContextManager`

```suggestion
def atomic_open(filename: StrOrBytesPath) -> _GeneratorContextManager[BufferedWriter]: ...
```

Where `StrOrBytesPath` is imported from `_typeshed`